### PR TITLE
fix/PRO-2387/update-prime-sdk

### PR DESCRIPTION
**Pull request by RanaBug**
## Description

- Update of the Prime SDK version in Transaction Kit, with 3 new added functions 'getQuotes', 'getTransactionStatus' and 'getSupportedAssets'

## How Has This Been Tested?

- These functions have been tested in the App example, and also using JEST

## Types of changes

- [x ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by RanaBug** [comment]
In progress...

File: src/hooks/useEtherspotHistory.ts [hunk 1]
```diff
@@ -59,24 +64,47 @@ const useEtherspotHistory = (chainId?: number): IEtherspotHistoryHook => {
 
   const getAccountTransaction = async (
     hash: string,
-    historyChainId: number = defaultChainId,
+    historyChainId: number = defaultChainId
   ): Promise<Transaction | undefined> => {
     try {
       const dataService = getDataService();
+      console.log('dataservice getAccountTransaction', dataService);
```
Comment (IAmKio) [inline src/hooks/useEtherspotHistory.ts:71]: Ooops, no informational console logs here please (it pollutes the developer's developer console)

Comment (IAmKio) [reply src/hooks/useEtherspotHistory.ts:71]: I think i added this actually and forgot to remove it, apologies

File: src/hooks/useEtherspotHistory.ts [hunk 2]
```diff
@@ -59,24 +64,47 @@ const useEtherspotHistory = (chainId?: number): IEtherspotHistoryHook => {
...
+    fromChainId: number,
+    toChainId: number,
+    hash: string,
+    provider?: BridgingProvider
+  ): Promise<TransactionStatus | undefined> => {
+    try {
+      const dataService = getDataService();
+      return dataService.getTransactionStatus({ fromChainId, toChainId, transactionHash: hash, provider });
+    } catch (e) {
+      console.error(e);
```
Comment (IAmKio) [inline src/hooks/useEtherspotHistory.ts:92]:
```suggestion
```
Sorry i also added this to try and tease out the error when we were debugging this earlier!

File: src/hooks/useEtherspotSwaps.ts [hunk 3]
```diff
@@ -116,17 +123,57 @@ const useEtherspotSwaps = (chainId?: number): IEtherspotSwapsHook => {
...
+        provider,
+      });
+
+      return quotes;
+    } catch (e) {
+      console.warn(
+        `Sorry, an error occurred whilst trying to fetch cross-chain quotes.` + ` Please try again. Error:`,
+        e
       );
     }
```
Comment (IAmKio) [inline src/hooks/useEtherspotSwaps.ts:169]: Unless i'm missing something we should return an empty array here in the case of an error, or throw an error

File: __tests__/hooks/useEtherspotHistory.test.js [hunk 4]
```diff
@@ -71,4 +63,54 @@ describe('useEtherspotHistory()', () => {
...
+
+    // missing props
+    const transaction3 = await result.current
+      .getAccountTransactionStatus(4, 137)
+      .catch((e) => {
+        console.error(e);
+        return e;
+      });
+
+    expect(transaction3).toBeNull()
```
Comment (IAmKio) [inline __tests__/hooks/useEtherspotHistory.test.js:113]: If we're missing required properties here, should we not return an error to let the developer know they have missed something (as opposed to `null`)?

**Review by IAmKio — CHANGES_REQUESTED** [review]
Absolutely fantastic work - just a few suggestions and minor code cleanup and we're good to go 🚀

File: src/hooks/useEtherspotHistory.ts (hunk 1 above)
Comment (RanaBug) [reply src/hooks/useEtherspotHistory.ts:71]: Oops good catch!

File: src/hooks/useEtherspotSwaps.ts (hunk 3 above)
Comment (RanaBug) [reply src/hooks/useEtherspotSwaps.ts:169]: should we not warn that there is an error though? if empty array, the user will think that there is no quotes for what they are asking for, rather being an error?

File: __tests__/hooks/useEtherspotHistory.test.js (hunk 4 above)
Comment (RanaBug) [reply __tests__/hooks/useEtherspotHistory.test.js:113]: Yes I've changed that now, I kept it simple

File: src/hooks/useEtherspotSwaps.ts (hunk 3 above)
Comment (IAmKio) [reply src/hooks/useEtherspotSwaps.ts]: Sorry just to be clear - i meant both return an empty array **in addition to** the warning

File: src/hooks/useEtherspotSwaps.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { useMemo } from 'react';
 // types
 import { ICrossChainSwapOffers, ISameChainSwapOffers } from '../types/EtherspotTransactionKit';
 
-// hopoks
```
Comment (IAmKio) [inline src/hooks/useEtherspotSwaps.ts:9]: lol

**Review by IAmKio — APPROVED** [review]
LGTM 🚀

**Pull request merged**